### PR TITLE
Refactor CI, tests and enable CD (publish to PYPI on tag)

### LINKED ISSUE #10: Setup automated deployment
There is seems to be a bug in travis, https://github.com/aio-libs/aiocassandra/blob/master/.travis.yml#L39

deploy section uses system provided Python2.7 with broken SNI module and it fails hard

### PR DESCRIPTION
**Pull request by hellysmile**
Resolves #10

### CODE REVIEW

File: aiocassandra.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 except ImportError:
     import trollius as asyncio
 
-__version__ = '1.1.0'
+__version__ = '1.1.1dev2'
```
Comment (webknjaz) [inline aiocassandra.py]: @hellysmile `setup.get_version()` is unable to correctly parse this :(

Comment (hellysmile) [reply aiocassandra.py]: Nice catch!

File: .travis.yml [hunk 2]
```diff
@@ -9,11 +9,15 @@ python:
...
+env:
+  matrix:
+    - CASSANDRA_VERSION=311x CQL_VERSION=3.4.4
+
 addons:
   apt:
     sources:
     - sourceline: 'ppa:webupd8team/java'
-    - sourceline: 'deb http://www.apache.org/dist/cassandra/debian 311x main'
+    - sourceline: 'deb http://www.apache.org/dist/cassandra/debian $CASSANDRA_VERSION main'
```
Comment (webknjaz) [inline .travis.yml]: Interpolation does not work here.

File: tox.ini [hunk 3]
```diff
@@ -27,7 +28,7 @@ deps =
 whitelist_externals =
     bash
     echo
-setenv =
+passenv =
     CQL_VERSION=3.4.4
```
Comment (webknjaz) [inline tox.ini]: you should delete the value in this case

File: .travis.yml [hunk 4]
```diff
@@ -28,6 +25,7 @@ services:
 
 before_install:
 - java -version
+- python -V
```
Comment (webknjaz) [inline .travis.yml]: you don't need this, travis already outputs python version anyway: https://travis-ci.org/aio-libs/aiocassandra/jobs/340881258#L538

File: .travis.yml [hunk 5]
```diff
@@ -1,51 +1,59 @@
...
+- cassandra
+
 before_install:
-  - sudo add-apt-repository -y ppa:webupd8team/java
-  - echo "deb http://www.apache.org/dist/cassandra/debian $CASSANDRA_VERSION main" | sudo tee -a /etc/apt/sources.list.d/cassandra.sources.list
-  - echo "deb-src http://www.apache.org/dist/cassandra/debian $CASSANDRA_VERSION main" | sudo tee -a /etc/apt/sources.list.d/cassandra.sources.list
-  - curl https://www.apache.org/dist/cassandra/KEYS | sudo apt-key add -
-  - sudo apt-get update -y
+- java -version
+- python -V
```
Comment (webknjaz) [inline .travis.yml]: this is redundant

File: tox.ini [hunk 6]
```diff
@@ -8,16 +8,16 @@ deps =
...
-        isort --check-only --diff $TESTS_LIST                                       \
-    '
+    bash -c "                                                   \
+        flake8 --show-source aiocassandra.py setup.py tests.py; \
+        isort --check-only aiocassandra.py setup.py tests.py;   \
+    "
 
-    ./tests.py
+    # envpython to mute "WARNING:test command found but not installed in testenv"
+    {envpython} tests.py
```
Comment (webknjaz) [inline tox.ini]: better rely on env python detection of tox. tests should be refactored  anyway

Comment (webknjaz) [reply tox.ini]: mind replacing it w/ pytest?

Comment (hellysmile) [reply tox.ini]: yes, makes perfect sense

**Pull request merged**